src/puppetlabs/ezbake/core.clj

```diff
@@ -189,9 +189,38 @@
       (.substring filename 0 (- (.length filename) 4))
       filename)))
 
-(defn quoted-list
-  [l]
-  (if (empty? l) "" (format "\"%s\"" (str/join "\",\"" l))))
+(defprotocol AsRubyLiteral
+  "Converts item to a corresponding, properly escaped ruby literal.
+  e.g. [\"x\" \"'\"] becomes \"['x', '\\'']\"."
+  (as-ruby-literal [this]))
+
+(extend-protocol AsRubyLiteral
+  ;; No floating point for now since we don't need it and haven't
+  ;; checked the syntax correspondence carefully yet.
+
+  nil (as-ruby-literal [s] "nil")
+
+  Boolean (as-ruby-literal [s] (str s))
+
+  Byte (as-ruby-literal [s] (str s))
+  Short (as-ruby-literal [s] (str s))
+  Integer (as-ruby-literal [s] (str s))
+  Long (as-ruby-literal [s] (str s))
+  BigInteger (as-ruby-literal [s] (str s))
+  clojure.lang.BigInt (as-ruby-literal [s] (str s))
+
+  String
+  (as-ruby-literal [s]
+    (let [esc (-> s
+                  (str/replace "\\" "\\\\")
+                  (str/replace "'" "\\'"))]
+      (str "'" esc "'")))
```

Comment:
Some of the ruby values in `ezbake.rb.moustache` have single quotes and some have double quotes. Double quotes have different interpolation rules than single quotes. Do you know if that matters?

...iow, do we ever pass set a string in our clojure ezbake configuration that we expect to be interpolated within Ruby?

Reply:
Ahh, that behavior is intentional (finally reading the commit).

Reply:
I think the concern is valid (though I hope we never do that anywhere) and slightly different than the point of the PR. I've double checked and I don't think[pe-]puppetserver isn't using such tomfoolery.